src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -35,8 +34,8 @@ def initialize(server, port, options = {})
 
     @server = server
     @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+    self.class.use_ssl = options[:use_ssl]
```

Comment:
@camlow325 Hmm, that's a good point. Since we're having a static underlying client object, that client can't change across instances of the `HttpClient` class, so my guess would be the client would try to make an https call without configuring an SSLContext. 

I'm wondering how we could handle this. Maybe we could regenerate the client whenever the value of `use_ssl` is changed?